scripts/rke2-uninstall.sh

```diff
@@ -0,0 +1,135 @@
+        INSTALL_RKE2_ROOT=${STYLUS_ROOT}/usr
+    else
+        INSTALL_RKE2_ROOT=${STYLUS_ROOT}/usr/local
+    fi 
+# Check if the OS is SUSE
+elif [ "${ID_LIKE%%[ ]*}" = "suse" ]; then 
+    if rpm -q rke2-common >/dev/null 2>&1; then
+        INSTALL_RKE2_ROOT=${STYLUS_ROOT}/usr
+        if [ -x /usr/sbin/transactional-update ]; then
+            transactional_update="transactional-update -c --no-selfupdate -d run"
+        fi
+    elif check_target_mountpoint "${STYLUS_ROOT}/usr/local" || check_target_ro "${STYLUS_ROOT}/usr/local"; then
+        INSTALL_RKE2_ROOT=${STYLUS_ROOT}/opt/rke2
+    else
+        INSTALL_RKE2_ROOT=${STYLUS_ROOT}/usr/local
+    fi
+# Default to /usr for other OSes
+else
+    INSTALL_RKE2_ROOT=${STYLUS_ROOT}/usr 
+fi
+
+# Uninstall killall script
+uninstall_killall() {
+    _killall="$(dirname "$0")/rke2-killall.sh"
+    if [ -e "${_killall}" ]; then
+        eval "${_killall}"
+    fi
+}
+
+# Disable services
+uninstall_disable_services() {
+    if command -v systemctl >/dev/null 2>&1; then
+        systemctl disable rke2-server || true
+        systemctl disable rke2-agent || true
+        systemctl reset-failed rke2-server || true
+        systemctl reset-failed rke2-agent || true
+        systemctl daemon-reload
+    fi
+}
+
+# Remove files
+uninstall_remove_files() {
+    if [ -r /etc/redhat-release ] || [ -r /etc/centos-release ] || [ -r /etc/oracle-release ] || [ -r /etc/amazon-linux-release ]; then
+        yum remove -y "rke2-*"
+        rm -f ${STYLUS_ROOT}/etc/yum.repos.d/rancher-rke2*.repo
+    fi
+
+    if [ "${ID_LIKE%%[ ]*}" = "suse" ]; then
+        if rpm -q rke2-common >/dev/null 2>&1; then
+            uninstall_cmd="zypper remove -y rke2-server rke2-agent rke2-common rke2-selinux"
+            if [ "${TRANSACTIONAL_UPDATE=false}" != "true" ] && [ -x /usr/sbin/transactional-update ]; then
+                uninstall_cmd="transactional-update -c --no-selfupdate -d run $uninstall_cmd"
+            fi
+            $uninstall_cmd
+            rm -f ${STYLUS_ROOT}/etc/zypp/repos.d/rancher-rke2*.repo
+        fi
+    fi
+
+    $transactional_update find "${INSTALL_RKE2_ROOT}/lib/systemd/system" -name rke2-*.service -type f -delete
+    $transactional_update find "${INSTALL_RKE2_ROOT}/lib/systemd/system" -name rke2-*.env -type f -delete
+    find ${STYLUS_ROOT}/etc/systemd/system -name rke2-*.service -type f -delete
+    $transactional_update rm -f "${INSTALL_RKE2_ROOT}/bin/rke2"
+    $transactional_update rm -f "${INSTALL_RKE2_ROOT}/bin/rke2-killall.sh"
+    $transactional_update rm -rf "${INSTALL_RKE2_ROOT}/share/rke2"
+    
+    # Removing directories with STYLUS_ROOT support
+    rm -rf ${STYLUS_ROOT}/etc/rancher/rke2
+    rm -rf ${STYLUS_ROOT}/etc/rancher/node
+    rm -d ${STYLUS_ROOT}/etc/rancher || true
+    rm -rf ${STYLUS_ROOT}/etc/cni
+    rm -rf ${STYLUS_ROOT}/opt/cni/bin
+    rm -rf ${STYLUS_ROOT}/var/lib/kubelet || true
+    rm -rf "${RKE2_DATA_DIR}"
```

Comment:
@gabelchinmay what is this path? does it need to be prefixed?

Reply:
it is "/var/lib/rancher/rke2"